syntax = "proto3";

option java_multiple_files = true;
option java_package = "api.DahuaSdkUltimate";
option java_outer_classname = "DahuaSdkUltimateProto";
option objc_class_prefix = "DAHUA_SDK_ULTIMATE";

package dahua_sdk_ultimate;

// The greeting service definition.
service DahuaManageService {
  // Sends a greeting
  rpc SayHello (HelloRequest) returns (HelloReply) {}

  rpc SayHelloStreamReply (HelloRequest) returns (stream HelloReply) {}

  rpc SayHelloBidiStream (stream HelloRequest) returns (stream HelloReply) {}
}

// The request message containing the user's name.
message HelloRequest {
  string name = 1;
}

// The response message containing the greetings
message HelloReply {
  string message = 1;
}
